helix-view/src/document.rs

```diff
@@ -888,7 +888,20 @@ impl Document {
                 if let Ok(metadata) = fs::metadata(&path).await {
                     if let Ok(mtime) = metadata.modified() {
                         if last_saved_time < mtime {
-                            bail!("file modified by an external process, use :w! to overwrite");
+                            let mut external_hasher = blake3::Hasher::default();
+                            let mut editor_hasher = blake3::Hasher::default();
+
+                            external_hasher.update_mmap_rayon(&path)?;
+                            text.chunks().for_each(|chunk| {
```

Comment:
This doesn't make sense. If you just want to check if the file on disk is the same on as the one in memory you can just do a comparison but that isnt what we should do here. 

Instead you need to compute the hash when the file is loaded from disk so you can check whether the file in memory right now is the same as the one we read from disk. Like I said everywhere we read the timestamp we also need to compute the hash (if the timestamp changed)

Reply:
Is there a good reason *not* to just directly compare text contents here, avoiding the hash altogether?  Because I can't think of one.

Either way we have to load the file contents from disk (either to compute the hash or to compare directly).  And computing the hash is likely(?) more expensive than just a direct equality check of the actual contents.

A hash is certainly useful to have around, but I think that's more for things like persistent undo where you don't want to store a whole copy of the file with the on-disk undo stack just to verify the stack's validity.  So it's more of a space-saving measure, there.

Reply:
Ignore the current implementation here that does not reflect what I had in mind. I left a review about that.

The hash is supposed to be a backup to mtime to avoid false positives. The hash will be computed when the file is read and the compared when the file is written. So we can't compare text directly because the text will have been edited since the file was read. You could keep a rope of the old file around but that would probably increase the memory consumption too much if you edit a lot of/very large files.

So it's a memory saving measure so we don't have to keep a copy of the filet hat was read form disk around. Vim does that too afaik

Reply:
> So we can't compare text directly because the text will have been edited since the file was read.

Ah!  Right.  Brain fart on that, ha ha.  Yeah, you're absolutely right that a hash is a good approach for this.